{"version":3,"file":"index.js","sourceRoot":"","sources":["../src/index.ts"],"names":[],"mappings":";;;;;;;;;;;;;;;AAAA,sCAAwC;AACxC,6BAA+B;AAE/B;;GAEG;AACU,QAAA,cAAc,GAAG,8EAA8E,CAAC;AAE7G;;GAEG;AACH;IAA+B,6BAAmB;IAChD,mBAAY,OAAgB,EAAS,KAAa;QAAlD,YACE,kBAAM,OAAO,CAAC,SACf;QAFoC,WAAK,GAAL,KAAK,CAAQ;;IAElD,CAAC;IAED,oBAAC,cAAO,CAAC,MAAM,IAAI,SAAS,CAAC,GAA7B;QACE,OAAO,SAAS,CAAC,IAAI,CAAC,CAAC;IACzB,CAAC;IACH,gBAAC;AAAD,CAAC,AARD,CAA+B,SAAS,CAAC,SAAS,GAQjD;AARY,8BAAS;AAUtB;;GAEG;AACH,SAAgB,SAAS,CAAC,KAAwB;IAChD,IAAI,GAAG,GAAuB,KAAmB,CAAC,KAAK,CAAC;IACxD,IAAI,SAAS,GAAG,KAAK,CAAC,KAAK,IAAI,KAAK,CAAC,OAAO,IAAI,EAAE,CAAC;IAEnD,OAAO,GAAG,EAAE;QACV,SAAS,IAAI,sBAAc,CAAC;QAC5B,SAAS,IAAI,GAAG,CAAC,KAAK,IAAI,GAAG,CAAC,OAAO,IAAI,EAAE,CAAC;QAE5C,GAAG,GAAI,GAAiB,CAAC,KAAK,CAAC;KAChC;IAED,OAAO,SAAS,CAAC;AACnB,CAAC;AAZD,8BAYC","sourcesContent":["import * as makeError from \"make-error\";\nimport { inspect } from \"util\";\n\n/**\n * @internal\n */\nexport const SEPARATOR_TEXT = `\\n\\nThe following exception was the direct cause of the above exception:\\n\\n`;\n\n/**\n * Create a new error instance of `cause` property support.\n */\nexport class BaseError extends makeError.BaseError {\n  constructor(message?: string, public cause?: Error) {\n    super(message);\n  }\n\n  [inspect.custom || \"inspect\"]() {\n    return fullStack(this);\n  }\n}\n\n/**\n * Capture the full stack trace of any error instance.\n */\nexport function fullStack(error: Error | BaseError) {\n  let err: Error | undefined = (error as BaseError).cause;\n  let fullStack = error.stack || error.message || \"\";\n\n  while (err) {\n    fullStack += SEPARATOR_TEXT;\n    fullStack += err.stack || err.message || \"\";\n\n    err = (err as BaseError).cause;\n  }\n\n  return fullStack;\n}\n"]}